### ISSUE TITLE
Preliminary Noetic Roadmap

### ISSUE THREAD

**Issue by matthew-reynolds** [ros-controls/ros_control#420]
Just want to start a place to collect changes for Noetic ahead of its release, since there doesn't seem to be a tracking issue yet. Happy to close in favour of a more formal roadmap if a maintainer makes one.

 - Do not release `_tests` packages (Related: #409)
 - Replace enums with enum classes (#412)
 - Remove non-const `ForceTorqueSensorHandle` constructor previously left in for ABI compatibility. Probably should have been removed in Melodic, but it got missed (#437)
 - Rename `hardware_interface::ResourceManager::resource_manager_type` to `ResourceManagerType` (#438)
 - Switch to `setuptools` (#429)

**Comment by bmagyar** [comment]
* Review all Python code and ensure Python3 compatibility

**Comment by sloretz** [comment]
* Bump CMake version to avoid CMP0048 warning on ROS build farm #427 

**Edit**: Also, all dependencies of packages in this repo have been released into ROS Noetic, so it can be released as soon as it's ready.

**Comment by sloretz** [comment]
Friendly ping, how's progress on a ROS Noetic release? It looks like all of `ros_control`'s dependencies are released, and [it's currently blocking 110 repos](http://repositories.ros.org/status_page/blocked_releases_noetic.html?q=id%3D%22ros_control%22&s=6&r=1).

**Comment by matthew-reynolds** [comment]
I think the main blocker is #429, which is waiting on the next Catkin release (https://github.com/ros/rosdistro/pull/24386)

**Comment by bmagyar** [comment]
CI is coming alive:
https://travis-ci.org/github/ros-controls/ros_control/jobs/675379090

**Comment by bmagyar** [comment]
https://github.com/ros-controls/ros_control/pull/439

**Comment by bmagyar** [comment]
A follow-up ticket for after the release: https://github.com/ros-controls/ros_control/issues/448

**Comment by bmagyar** [comment]
@sloretz  you can find the initial release here: https://github.com/ros/rosdistro/pull/24541
@matthew-reynolds I'll create a new issue for the next release, a few non-breaking PR-s we should add to noetic soonish

**Comment by bmagyar** [comment]
https://github.com/ros-controls/ros_control/issues/449
